src/Akka.Persistence.Azure/Snapshot/AzureBlobSnapshotStore.cs

```diff
@@ -43,26 +54,42 @@ public AzureBlobSnapshotStore(Config config = null)
             _storageAccount = _settings.Development ? 
                 CloudStorageAccount.DevelopmentStorageAccount : 
                 CloudStorageAccount.Parse(_settings.ConnectionString);
+
             _container = new Lazy<CloudBlobContainer>(() => InitCloudStorage().Result);
         }
 
         public CloudBlobContainer Container => _container.Value;
 
         private async Task<CloudBlobContainer> InitCloudStorage()
         {
-            var blobClient = _storageAccount.CreateCloudBlobClient();
-            var containerRef = blobClient.GetContainerReference(_settings.ContainerName);
-            var op = new OperationContext();
+            var retry = 5;
+            var exceptions = new List<Exception>();
 
-            using (var cts = new CancellationTokenSource(_settings.ConnectTimeout))
+            while (true)
             {
-                if (await containerRef.CreateIfNotExistsAsync(BlobContainerPublicAccessType.Container,
-                    new BlobRequestOptions(), op, cts.Token))
-                    _log.Info("Created Azure Blob Container", _settings.ContainerName);
-                else
-                    _log.Info("Successfully connected to existing container", _settings.ContainerName);
+                try
+                {
+                    var blobClient = _storageAccount.CreateCloudBlobClient();
+                    var containerRef = blobClient.GetContainerReference(_settings.ContainerName);
+                    var op = new OperationContext();
+
+                    if (await containerRef.CreateIfNotExistsAsync(BlobContainerPublicAccessType.Container, new BlobRequestOptions(), op))
+                        _log.Info("Created Azure Blob Container", _settings.ContainerName);
+                    else
+                        _log.Info("Successfully connected to existing container", _settings.ContainerName);
 
-                return containerRef;
+                    return containerRef;
+                }
+                catch (Exception ex)
+                {
+                    retry--;
+                    if (retry < 0)
```

Comment:
This should be `retry == 0` - otherwise your dictionary accessor attempt will fail with a `MissingKeyException` below`